mmcv/runner/base_runner.py

```diff
@@ -69,7 +69,11 @@ def __init__(self,
                     'batch_processor and model.train_step()/model.val_step() '
                     'cannot be both available.')
         else:
-            assert hasattr(model, 'train_step')
+            if is_parallel_module(model):
+                _model = model.module
+            else:
+                _model = model
+            assert hasattr(_model, 'train_step')
```

Comment:
- MMDDP has already implemented `train_step()`.
- In the runner, `model.train_step()` is called instead of `model.module.train_step()`.

Reply:
MMDP doesn't have `train_step()` and `val_step()` method, which will raise error in some non-distributed training. So I add these two methods in MMDP like those in MMDDP.